### PR TITLE
Some modernization of plugins in EgammaIsolationAlgos

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Employ the modern ways to get ED products in EgammaIsolationAlgos plugins and some other changes.

#### PR validation:

CMSSW compiles.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport intended.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: RecoEgamma/EgammaIsolationAlgos/plugins/EgammaEcalRecHitIsolationProducer.cc [hunk 1]
```diff
@@ -100,25 +90,16 @@ EgammaEcalRecHitIsolationProducer::EgammaEcalRecHitIsolationProducer(const edm::
...
 void EgammaEcalRecHitIsolationProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
   // Get the  filtered objects
-  edm::Handle<edm::View<reco::Candidate>> emObjectHandle;
-  iEvent.getByLabel(emObjectProducer_, emObjectHandle);
+  auto emObjectHandle = iEvent.getHandle(emObjectProducer_);
 
   // Next get Ecal hits barrel
-  edm::Handle<EcalRecHitCollection> ecalBarrelRecHitHandle;  //EcalRecHitCollection is a typedef to
-  iEvent.getByLabel(ecalBarrelRecHitProducer_.label(), ecalBarrelRecHitCollection_.label(), ecalBarrelRecHitHandle);
+  auto ecalBarrelRecHitHandle = iEvent.getHandle(ecalBarrelRecHitCollection_);
```
Comment (slava77) [inline RecoEgamma/EgammaIsolationAlgos/plugins/EgammaEcalRecHitIsolationProducer.cc:99]: it's nice to see this odd pattern go away.
apparently this is the only case in the active code, based on `git grep "getByLabel[^$]*label()[^$]*label()"`

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/32651	64935f2
- technical; code changes are in line with the PR description 
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by qliphy** [comment]
+1

**Pull request merged**